### PR TITLE
VE-2354 Fix: Overflow & Cutoff on Right Edge

### PR DESCRIPTION
**Pull request by amey-shrivastava**
https://contentstack.atlassian.net/browse/VE-2354

### CODE REVIEW

File: src/styles.css [hunk 1]
```diff
@@ -1,3 +1,7 @@
+body {
+    overflow-x: hidden;
+}
```
Comment (faraazb) [inline src/styles.css:3]: We cannot do this. This will affect the user's website. We can increase the buffer if needed.

**Review by faraazb — CHANGES_REQUESTED** [review]

File: src/liveEditor/generators/generateToolbar.tsx [hunk 2]
```diff
@@ -66,8 +67,21 @@ export function appendFieldPathDropdown(
         targetElementDimension.left - LIVE_PREVIEW_OUTLINE_WIDTH_IN_PX;
     const adjustedDistanceFromLeft = Math.max(distanceFromLeft, TOOLBAR_EDGE_BUFFER);
 
+    const targetElementRightEdgeOffset =
+        window.innerWidth - targetElementDimension.left;
+
+    if (targetElementRightEdgeOffset < RIGHT_EDGE_BUFFER) {
```
Comment (faraazb) [inline src/liveEditor/generators/generateToolbar.tsx:73]: We are also not considering the available `window.scrollX` space that may be available like we do for stop edge.

**Review by faraazb — CHANGES_REQUESTED** [review]

File: src/styles.css (hunk 1 above)
Comment (amey-shrivastava) [reply src/styles.css:3]: ok I will consider this as well. Will update code

File: src/liveEditor/generators/generateToolbar.tsx (hunk 2 above)
Comment (amey-shrivastava) [reply src/liveEditor/generators/generateToolbar.tsx:73]: ok I will consider this as well. Will update code

**Review by faraazb — APPROVED** [review]
Looks good to me!

**Pull request merged**